t/porting/libperl.t

```diff
@@ -211,6 +211,14 @@ sub nm_parse_gnu {
     } else {
         die "$0: undefined current object: $line"
             unless defined $symbols->{o};
+
+        # generate a cross-ref of every line each symbol appears on,
+        # for diagnostics
+        for my $sym (grep !/^[0-9A-Fa-f]+$/, /\b(\w{2,})\b/g) {
```

Comment:
`[0-9A-Fa-f]` could be written as `[[:xdigit:]]`, which I like better.
`$` is semantically wrong; should be `\z`.
But I'd probably invert the whole thing and write `grep /[^[:xdigit:]]/, ...` ("must contain a non-hex-digit").

(The second regex could be just `/\w{2,}/g`, but the explicit form is fine, too.)